runtime/freelist.c

```diff
@@ -460,13 +461,13 @@ header_t *caml_fl_merge_block (value bp)
   cur = Next (prev);
   /* The sweep code makes sure that this is the right place to insert
      this block: */
-  CAMLassert (prev < bp || prev == Fl_head);
-  CAMLassert (cur > bp || cur == Val_NULL);
+  CAMLassert (Bp_val (prev) < Bp_val (bp) || prev == Fl_head);
+  CAMLassert (Bp_val (cur) > Bp_val (bp) || cur == Val_NULL);
 
   if (policy == Policy_first_fit) truncate_flp (prev);
 
   /* If [last_fragment] and [bp] are adjacent, merge them. */
-  if (last_fragment == Hp_bp (bp)){
+  if (last_fragment == Hp_val (bp)){
```

Comment:
I think this is just a philosophical change reflecting that you want a pointer to the header rather than a pointer to the first byte specifically?

Reply:
You're right that it's philosophical, but `bp` is declared as a value and all the other macros applied to it are `_val`, so why not this one?

Reply:
Oh, I see - that was kind of where my question was coming from. So the idea is `Hp_bp` should be being given a pointer, not a `value`. Gotcha! The other reason for noting it was because it's a change which isn't in an assertion.